### PR TITLE
Fix handling of headers in HttpClient

### PR DESCRIPTION
**Pull request by jplatte**
I fully broke authentication in #83, here is the fix.

I'm sorry for yet again submitting a PR that conflicts with yours @DevinR528, but this should probably be merged first, making #84 smaller in diff too.

### CODE REVIEW

File: matrix_sdk/src/http_client.rs [hunk 1]
```diff
@@ -63,35 +64,14 @@ impl HttpClient {
...
-        };
+        if let HttpMethod::POST | HttpMethod::PUT | HttpMethod::DELETE = *request.method() {
+            request.headers_mut().append(
+                http::header::CONTENT_TYPE,
+                HeaderValue::from_static("application/json"),
+            );
+        }
 
-        Ok(request_builder.send().await?)
+        Ok(self.inner.execute(request.try_into()?).await?)
```
Comment (poljar) [inline matrix_sdk/src/http_client.rs:74]: This method doesn't seem to exist for the WASM version of `reqwest` so the CI fails under WASM.

Comment (jplatte) [reply matrix_sdk/src/http_client.rs:74]: Fixed two days ago in https://github.com/seanmonstar/reqwest/pull/989, seems to just require another `reqwest` release. I don't think there's a good workaround, should I maybe just update reqwest to a git dependency for now?

Comment (poljar) [reply matrix_sdk/src/http_client.rs:74]: I guess so, since we're already blocked by a bunch of git deps adding another one won't hurt that much.

**Comment by jplatte** [comment]
Blocked on https://github.com/seanmonstar/reqwest/issues/996.

**Pull request merged**